governator-core/src/main/java/com/netflix/governator/InjectorBuilder.java

```diff
@@ -0,0 +1,221 @@
+ *      .createInjector();
+ * </code>
+ */
+public final class InjectorBuilder {
+    private static final Logger LOG = LoggerFactory.getLogger(InjectorBuilder.class);
+    
+    private static final Stage LAZY_SINGLETONS_STAGE = Stage.DEVELOPMENT;
+    
+    private Module module;
+    
+    /**
+     * Start the builder using the specified module. 
+     * 
+     * @param module
+     * @return
+     */
+    public static InjectorBuilder fromModule(Module module) {
+        return new InjectorBuilder(module);
+    }
+    
+    public static InjectorBuilder fromModules(Module firstModule, Module secondModule, Module ... additionalModules) {
+        List<Module> modules = new ArrayList<>();
+        modules.add(firstModule);
+        modules.add(secondModule);
+        if (null != additionalModules) {
+            modules.addAll(Arrays.asList(additionalModules));
+        }
+        return new InjectorBuilder(Modules.combine(modules));
+    }
+    
+    public static InjectorBuilder fromModules(List<Module> modules) {
+        return new InjectorBuilder(Modules.combine(modules));
+    }
+    
+    public static InjectorBuilder createDefault() {
+        return fromModule(new Module() {
+            @Override
+            public void configure(Binder binder) {
+            }
+        });
+    }
+    
+    private InjectorBuilder(Module module) {
+        this.module = module;
+    }
+    
+    /**
+     * Override all existing bindings with bindings in the provided modules.
+     * This method uses Guice's build in {@link Modules#override} and is preferable
+     * to using {@link Modules#override}.  The approach here is to attempt to promote 
+     * the use of {@link Modules#override} as a single top level override.  Using
+     * {@link Modules#override} inside Guice modules can result in duplicate bindings 
+     * when the same module is installed in multiple placed. 
+     * @param modules
+     */
+    public InjectorBuilder overrideWith(Module ... modules) {
+        return overrideWith(Arrays.asList(modules));
+    }
+    
+    /**
+     * @see InjectorBuilder#overrideWith(Module...)
+     */
+    public InjectorBuilder overrideWith(Collection<Module> modules) {
+        this.module = Modules.override(module).with(modules);
+        return this;
+    }
+    
+    /**
+     * Add additional bindings to the module tracked by the DSL
+     * @param modules
+     */
+    public InjectorBuilder combineWith(Module ... modules) {
+        List<Module> m = new ArrayList<>();
+        m.add(module);
+        m.addAll(Arrays.asList(modules));
+        this.module = Modules.combine(m);
+        return this;
+    }
+    
+    /**
+     * Iterator through all elements and invoke a visitor.  This method
+     * is mean for side-effect free operations such as logging the current 
+     * state of bindings.
+     * @param visitor
+     */
+    public InjectorBuilder traceEachElement(ElementVisitor<String> visitor) {
```

Comment:
"trace" seems more specific than necessary here for the method name. wasn't it forEachElement? You could also try visitEachElement?